client.go

```diff
@@ -150,7 +152,9 @@ func (c *ClientImpl) getQueryParams(setStatements map[string]string) (map[string
 		if len(c.AccountID) == 0 {
 			return nil, fmt.Errorf("Trying to run a query against system engine without account id defined")
 		}
-		params["account_id"] = c.AccountID
+		if _, ok := params["account_id"]; !ok {
+			params["account_id"] = c.AccountID
+		}
```

Comment:
This will be controled by the Firebolt-Update-Endpoint header. Also, for V2 engines we always have `ConnectedToSystemEngine=true` since we have no way of knowing what engine is actually used. So basically, we just make sure here that is account_id is not set, we set it to the value stored